### PR TITLE
added tests

### PR DESCRIPTION
**Pull request by andrii-subash**

### CODE REVIEW

**Comment by andrii-subash** [comment]
image

File: src/test/java/book/store/controller/BookControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+import org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class BookControllerTest {
```
Comment (Ivan95kos) [inline src/test/java/book/store/controller/BookControllerTest.java:42]:
```suggestion
class BookControllerIntegrationTest {
```

File: src/test/java/book/store/controller/CategoryControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+import org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class CategoryControllerTest {
```
Comment (Ivan95kos) [inline src/test/java/book/store/controller/CategoryControllerTest.java:39]: the same

File: src/test/java/book/store/repository/BookRepositoryTest.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.test.context.jdbc.Sql;
+
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+class BookRepositoryTest {
```
Comment (Ivan95kos) [inline src/test/java/book/store/repository/BookRepositoryTest.java:16]: also

**Review by Ivan95kos — APPROVED** [review]
Great job, with minor comments

**Pull request merged**